src/master/services/ClusterMonitor.ts

```diff
@@ -13,43 +13,44 @@ export class ClusterMonitor extends IntervalService {
         super(10000, master.logger);
     }
 
-    public execute(cluster?: ClusterConnection): void {
-        if (cluster === undefined) {
-            return this.master.clusters.forEach((_, cluster) =>
-                cluster !== undefined ? this.execute(cluster) : undefined);
-        }
-
-        if (cluster.state !== WorkerState.RUNNING)
-            return;
+    public async execute(): Promise<void> {
+        await this.master.clusters.forEach((_, cluster) => this.#checkCluster(cluster));
+    }
 
-        const stats = this.master.clusterStats.get(cluster.id);
+    #getClusterIssues(cluster: ClusterConnection): string[] {
+        const stats = this.master.clusterStats.get(cluster);
         const now = moment();
         const cutoff = moment().add(-1, 'minute');
-        const alerts = [];
-
-        if (stats !== undefined) {
-            for (const shard of stats.shards) {
-                if (cutoff.isAfter(shard.time)) {
-                    const diff = moment.duration(now.diff(shard.time));
-                    const msg = `⏰ shard ${shard.id} unresponsive for ${diff.asSeconds()} seconds`;
-                    this.logger.cluster(msg);
-                    alerts.push(msg);
-                }
-            }
-        } else if (cluster.created.isBefore(cutoff)) {
-            const diff = moment.duration(now.diff(cluster.created));
-            const msg = `⏰ Cluster ${cluster.id} was created ${diff.asSeconds()} seconds ago but hasnt posted stats yet`;
-            this.logger.cluster(msg);
-            alerts.push(msg);
+
+        function secondsSince(time: number | moment.Moment): number {
```

Comment:
Potentially? I dont really want to make a `secondsSince` util method though because its not very flexible. I really only made this function here because the template strings were getting too long lol